dace/libraries/standard/nodes/gearbox.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+"""
+Provides a library node that converts from a stream of type
+vector(vector(dtype, w0)) to a stream of type vector(dtype, w1), or vice versa.
+This is useful for achieving efficient memory reads on Xilinx FPGAs, where
+modules accessing memories should always read or write 512-bit vectors, which
+then potentially need to be narrowed down to the vector width of the
+computational kernel.
+
+The node expects to have a single input and a single output, where one end is
+of type vector(vector(dtype, w0)), and the other is of type vector(dtype, w1).
+"""
+import collections
+import copy
+import dace
+
+
+@dace.library.expansion
+class ExpandGearbox(dace.transformation.ExpandTransformation):
+
+    environments = []
+
+    @staticmethod
+    def expansion(node: "Gearbox", parent_state: dace.SDFGState,
+                  parent_sdfg: dace.SDFG):
+
+        (in_edge, in_desc, out_edge, out_desc, is_pack,
+         gear_factor) = node.validate(parent_sdfg, parent_state)
+
+        if is_pack:
+            vtype = out_desc.dtype
+        else:
+            vtype = in_desc.dtype
+
+        sdfg = dace.SDFG("gearbox")
+        in_desc_inner = copy.copy(in_desc)
```

Comment:
I think it should be deepcopy because of location and stuff

Reply:
Done in da422396